### DISCUSSION TITLE
"segmentation fault" when starting development server. Anyone run into this problem?

### DISCUSSION THREAD

**Discussion by stylessard** [vercel/next.js#43436] [Help]
Hi everyone! After starting up a new Next project (using the automatic setup or manual setup), I keep getting "zsh: segmentation fault npm run dev" when trying to spin up the development server. I've searched around and tried suggestions like deleting node_modules and running npm install, updating node, re-installing node, but none have fixed the problem. Has anyone run into this problem and fixed it using some other approach? Thanks!

**Comment by icyJoseph** [comment]
What do you get if you do `npx next info`?

**Reply by stylessard** [reply]
I get this:
      
    Operatubg System
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 19.0.0: Wed Sep 25 20:18:50 PDT 2019; root:xnu-6153.11.26~2/RELEASE_X86_64
    Binaries:
      Node: 18.12.1
      npm: 8.19.2
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.0.6-canary.1
      eslint-config-next: 13.0.5
      react: 18.2.0
      react-dom: 18.2.0`

I also got a warning about installing the latest version of canary, which I did, but segmentation fault still appears.

**Reply by icyJoseph** [reply]
Alright I've got nothing :/ Tried with Node 18, but nada. I don't have a machine where I could try to reproduce this unfortunately.

Perhaps you could try with another shell? Do you get any other logs?

**Reply by stylessard** [reply]
I've tried with bash and zsh. Unfortunately, the same issue pops up with both. And no, no other logs, which is frustrating. I'll keep digging in. Thank you for your help!

**Comment by brianesa** [comment]
i've got the same issue. have tried with some version of node but still no changes

**Reply by stylessard** [reply]
Wish I had a solution to share, but nothing yet. I've seen a few more people pop up running into the same problem so I'm hoping the missing piece to the puzzle is figured out soon.

**Reply by icyJoseph** [reply]
Please also post `npx next info` we need to narrow this down as much as possible

**Reply by brianesa** [reply]
```
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 19.3.0: Thu Jan  9 20:58:23 PST 2020; root:xnu-6153.81.5~1/RELEASE_X86_64
    Binaries:
      Node: 18.12.1
      npm: 8.19.2
      Yarn: 1.22.17
      pnpm: N/A
    Relevant packages:
      next: 13.0.5
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

warn  - Latest canary version not detected, detected: "13.0.5", newest: "13.0.6-canary.3".
        Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
        Read more - https://nextjs.org/docs/messages/opening-an-issue
```

**Comment by icyJoseph** [comment]
You two, could you please try to use next.js with babel? Create a project and then create a .babelrc file. I'm not on my computer at the moment but you should be able to find online what has to go inside the Babel file. I want to discard SWC from the equation.

**Reply by icyJoseph** [reply]
I think this has to go inside `.babelrc` - the presence of babel makes next.js switch out from SWC.

```
{
  "presets": ["next/babel"]
}
```

**Reply by stylessard** [reply]
Still getting segmentation fault, but with this info now:

    info  - Disabled SWC as replacement for Babel because of custom Babel configuration
    info  - Using external babel configuration from .babelrc
    event - compiled client and server successfully in 6.3s (182 modules)
    zsh: segmentation fault  npm run dev

**Reply by icyJoseph** [reply]
Uff not good. You can run other node scripts correct? If you try to use CRA or vite-app? 

This is really weird. It ought to be related to your architecture, but what part is tripping, I'm puzzled about!

**Reply by stylessard** [reply]
It's strange. And yeah, I'm able to start up my CRA and Astro projects no problem. Just having issues with Next.

**Reply by brianesa** [reply]
i also face this issues when i use next `v12.2`

**Comment by AbhimanyuG** [comment]
I switched to last known working version on my machine

package.json

    "eslint-config-next": "12.3.1",
    "next": "12.3.1",

Remove dependencies @next/font

    //import { Inter } from '@next/font/google'

Remove inter.* references

    className={inter.className}

**Reply by stylessard** [reply]
This worked! Thank you so much! 

Do you know what is causing a select few of us to have this problem?

**Reply by AbhimanyuG** [reply]
Not sure, spent a lot of time. seg faults are difficult to debug.

**Reply by stylessard** [reply]
Well, way beyond appreciative! Thank you, again!

**Comment by rstarkov** [comment]
I'm getting this issue on Windows 10 and Node 18.15.0.

STR: just literally run the first command from the tutorial to create `nextjs-blog`, and `npm run dev` will fail right away, out of the box.

Output from Git Bash:
```
$ npm run build

> build
> next build

info  - Linting and checking validity of types
info  - Downloading WASM swc package...
info  - Using wasm build of next-swc
info  - Creating an optimized production build ./c/Apps/nodejs/npm: line 44:  1363 Segmentation fault      "$NODE_EXE" "$NPM_CLI_JS" "$@"
```

In cmd.exe it crashes without any errors whatsoever, which is rather unfortunate because it looks like it worked and there's absolutely nothing to google for!

In my case this happens for every 13.2.* version, but 13.1.6 works. Going into the 13.1.7 canaries, I found that the last working version was `13.1.7-canary.12`. There are only 4 commits in that range: [commits](https://github.com/vercel/next.js/commits/v13.1.7-canary.13) - maybe the turbopack upgrade?

Happy to run any further diagnostics, I just don't really know what else to try to get more info on the cause of this crash.

**Reply by icyJoseph** [reply]
What architecture do you have? 32 or 64 bits?

**Reply by rstarkov** [reply]
OS is 64 but `node -p "process.arch"` says ia32.

**Reply by icyJoseph** [reply]
Aha, is it possible you have node 32 bits installed?

**Reply by rstarkov** [reply]
@icyJoseph Indeed! Oops. Updated to node x64 and the build works correctly using the latest version of the `next` package. Thank you!
